redis/src/main/scala/zio/redis/TestExecutor.scala

```diff
@@ -972,6 +980,972 @@ private[redis] final class TestExecutor private (
           } yield RespValue.Array(Chunk.fromIterable(values))
         )
 
+//      case api.SortedSets.BzPopMax.name =>
+//        val keys = input.dropRight(1).map(_.asString)
+//
+//        orWrongType(forAll(keys)(isSortedSet))(
+//          (for {
+//            allSets <-
+//              STM.foreach(keys.map(key => STM.succeedNow(key) &&& sortedSets.getOrElse(key, SortedSet.empty)))(identity)
+//            nonEmptyLists <- STM.succeed(allSets.collect { case (key, v) if v.nonEmpty => key -> v })
+//            (sk, ss)      <- STM.fromOption(nonEmptyLists.headOption)
+//            max            = ss.last
+//            newSet         = ss.dropRight(1)
+//            _             <- sortedSets.put(sk, newSet)
+//          } yield Replies.array(Chunk(sk, max.member, max.score.toString)))
+//            .foldM(_ => STM.retry, result => STM.succeed(result))
+//        )
+//
+//      case api.SortedSets.BzPopMin.name =>
+//        val keys = input.dropRight(1).map(_.asString)
+//
+//        orWrongType(forAll(keys)(isSortedSet))(
+//          (for {
+//            allSets <-
+//              STM.foreach(keys.map(key => STM.succeedNow(key) &&& sortedSets.getOrElse(key, SortedSet.empty)))(identity)
+//            nonEmptyLists <- STM.succeed(allSets.collect { case (key, v) if v.nonEmpty => key -> v })
+//            (sk, ss)      <- STM.fromOption(nonEmptyLists.headOption)
+//            min            = ss.head
+//            newSet         = ss.drop(1)
+//            _             <- sortedSets.put(sk, newSet)
+//          } yield Replies.array(Chunk(sk, min.member, min.score.toString)))
+//            .foldM(_ => STM.retry, result => STM.succeed(result))
+//        )
+//
+      case api.SortedSets.ZAdd =>
+        val key = input(0).asString
+
+        val updateOption = input.map(_.asString).find {
+          case "XX" => true
+          case "NX" => true
+          case "LT" => true
+          case "GT" => true
+          case _    => false
+        }
+
+        val changedOption = input.map(_.asString).find {
+          case "CH" => true
+          case _    => false
+        }
+
+        val incrOption = input.map(_.asString).find {
+          case "INCR" => true
+          case _      => false
+        }
+
+        val optionsCount = updateOption.map(_ => 1).getOrElse(0) + changedOption.map(_ => 1).getOrElse(0) + incrOption
+          .map(_ => 1)
+          .getOrElse(0)
```

Comment:
Please use `fold` instead of `map` + `getOrElse`. This remark applies generally :).